dev/io.openliberty.microprofile.lra.1.0.internal/src/io/openliberty/microprofile/lra/internal/LraFilterComponent.java

```diff
@@ -0,0 +1,115 @@
+/*******************************************************************************
+ * Copyright (c) 2020 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package io.openliberty.microprofile.lra.internal;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+import com.ibm.websphere.ras.Tr;
+import com.ibm.websphere.ras.TraceComponent;
+import com.ibm.ws.jaxrs20.providers.api.JaxRsProviderRegister;
+
+import io.narayana.lra.client.NarayanaLRAClient;
+import io.narayana.lra.filter.ClientLRARequestFilter;
+import io.narayana.lra.filter.ClientLRAResponseFilter;
+import io.narayana.lra.filter.ServerLRAFilter;
+
+/**
+ * A declarative services component can be completely POJO based
+ * (no awareness/use of OSGi services).
+ *
+ * OSGi methods (activate/deactivate) should be protected.
+ */
+@Component(service = { JaxRsProviderRegister.class })
+public class LraFilterComponent implements JaxRsProviderRegister {
+
+    private static final TraceComponent tc = Tr.register(LraFilterComponent.class);
+
+    @Reference
+    private LraConfig config;
+
+    /**
+     * DS method to activate this component.
+     * Best practice: this should be a protected method, not public or private
+     *
+     * @param properties : Map containing service & config properties
+     *            populated/provided by config admin
+     */
+    @Activate
+    protected void activate(Map<String, Object> properties) throws LraException {
+        if (TraceComponent.isAnyTracingEnabled() && tc.isEventEnabled()) {
+            Tr.event(tc, "LraFilterComponent activated", properties);
+        }
+        String coordString = "http://" + config.getHost() + ":" + config.getPort() + "/" + config.getPath();
```

Comment:
Is this a http or https?

Reply:
http
https is unfortunately not yet supported in the implementation

Reply:
Should be configurable then?

Reply:
It will almost certainly need to be configurable once narayana supports https. However, as I don't know what their implementation will look like yet, I didn't think it was correct to add in liberty config that might have to change to fit in with the implementation